core/promise/consumer.go

```diff
@@ -32,17 +53,60 @@ func (c *consumer) NewRequest() (requestPtr interface{}) {
 }
 
 func (c *consumer) Consume(requestPtr interface{}) (response interface{}, err error) {
-	// request := requestPtr.(*Request)
+	request, ok := requestPtr.(*Request)
+	if !ok {
+		return nil, errUnsupportedRequest
+	}
+
+	receivedPromise, err := json.Marshal(request.SignedPromise.Promise)
+	if err != nil {
+		return nil, err
+	}
+
+	signature := identity.SignatureBase64(string(request.SignedPromise.IssuerSignature))
+	issuer := identity.FromAddress(request.SignedPromise.Promise.IssuerID)
+	verify := identity.NewVerifierIdentity(issuer)
+	if !verify.Verify(receivedPromise, signature) {
+		return &Response{
+			Success: false,
+			Message: errBadSignature.Error(),
+			Request: request,
+		}, nil
+	}
+
+	benefiter := identity.FromAddress(request.SignedPromise.Promise.BenefiterID)
+	if benefiter.Address != c.proposal.ProviderID {
+		return &Response{
+			Success: false,
+			Message: errUnknownBenefiter.Error(),
+			Request: request,
+		}, nil
+	}
+
+	price := c.proposal.PaymentMethod.GetPrice()
+	amount := request.SignedPromise.Promise.Amount.Amount
+	if amount < price.Amount {
+		return &Response{
+			Success: false,
+			Message: errLowAmount.Error(),
+			Request: request,
+		}, nil
+	}
 
-	// TODO there should be some validation of the received proposal and storing it somewhere for the server needs.
-	// TODO signature validation of the promise should be here too.
+	balance, err := c.etherClient.BalanceAt(context.Background(), common.HexToAddress(issuer.Address), nil)
+	if err != nil {
+		return nil, err
+	} else if balance.Uint64() < amount {
+		return &Response{
```

Comment:
this forming of responses seems to be a bit repetitive, maybe create a helper function like "ResponseFromError(err error, request)"

Reply:
I suggest static response with `const responseInsufficient = ..`